Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,72 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        private static int FindDigit(string equation)
+        {
+            double[] numbers = new double[3];
+            double result;
+            string unknown = "", res_string;
+            int missing_part = 0, index = 0, flag = 0;
+
+            string[] temp = equation.Split(new Char[] { '*', '=' });
+
+            for (int i = 0; i < 3; i++)
+            {
+                if (temp[i].IndexOf('?') < 0)
+                {
+                    string temp1 = temp[i];
+                    numbers[i] = Int32.Parse(temp1);
+                }
+                else
+                {
+                    unknown = temp[i];
+                    index = temp[i].IndexOf('?');
+                    missing_part = i;
+                }
+            }
+            if (missing_part == 0)
+            {
+                result = numbers[2] / numbers[1];
+                flag = Check(result);
+            }
+            else if (missing_part == 1)
+            {
+                result = numbers[2] / numbers[0];
+                flag = Check(result);
+            }
+            else
+            {
+                result = numbers[0] * numbers[1];
+                flag = Check(result);

```

Comment:
you can also try boolean flag so that extra check with number can be avoided and directly check for complaince